src/components/RangeSlider/components/SingleThumb/SingleThumb.scss

```diff
@@ -114,51 +173,38 @@ $range-thumb-shadow-focus: 0 0 0 rem(1px) color('indigo');
     transform: translateY($range-thumb-size / 5) scale(0.4);
   }
 
-  &::-webkit-slider-thumb {
-    margin-top: -($range-thumb-size - $range-track-height) / 2;
-  }
-
   .error & {
     --progress-lower: #{color('red')};
 
     @include range-track-selectors {
-      background-color: color('red', 'light');
+      background-color: var(--p-override-none, color('red', 'light'));
     }
 
     @include range-thumb-selectors {
-      border-color: color('red');
+      background: var(--p-critical, color('white'));
+      border-color: var(--p-critical, color('red'));
       box-shadow: $range-thumb-shadow-error;
     }
   }
 
   .disabled & {
     @include range-track-selectors {
       cursor: auto;
+      background-color: var(--p-border-on-inverse, color('sky', 'dark'));
```

Comment:
This feels wrong to me, for the same reasons as using the disabled variant for the track felt wrong in my earlier comment. This should use a disabled border variant in the neutral role, which @dleroux is working on creating. Sync with him and use something like that.